`System.Net.Mail.SmtpClient' is obsolete

Hello,

I'm trying to build with mono 4.6.2 on Debian, by running the command:

```
xbuild /p:Configuration=NET45-Release /t:RunAllTests buildscripts/Build.proj
```
This fails with
```
Build FAILED.
Errors:

/home/mans0954/2017/Core-4.0.0/buildscripts/Build.proj (RunAllTests) ->
(BuildProject target) ->
/home/mans0954/2017/Core-4.0.0/Castle.Core.sln (Build) ->
(Build target) ->
/home/mans0954/2017/Core-4.0.0/src/Castle.Core/Castle.Core.csproj (default targets) ->
/usr/lib/mono/xbuild/14.0/bin/Microsoft.CSharp.targets (CoreCompile target) ->

	Core/Smtp/DefaultSmtpSender.cs(233,47): error CS0618: Warning as Error: `System.Net.Mail.SmtpClient' is obsolete: `SmtpClient and its network of types are poorly designed, we strongly recommend you use https://github.com/jstedfast/MailKit and https://github.com/jstedfast/MimeKit instead'
	Core/Smtp/DefaultSmtpSender.cs(267,11): error CS0618: Warning as Error: `System.Net.Mail.SmtpClient' is obsolete: `SmtpClient and its network of types are poorly designed, we strongly recommend you use https://github.com/jstedfast/MailKit and https://github.com/jstedfast/MimeKit instead'
	Core/Smtp/DefaultSmtpSender.cs(159,5): error CS0618: Warning as Error: `System.Net.Mail.SmtpClient' is obsolete: `SmtpClient and its network of types are poorly designed, we strongly recommend you use https://github.com/jstedfast/MailKit and https://github.com/jstedfast/MimeKit instead'
	Core/Smtp/DefaultSmtpSender.cs(176,6): error CS0618: Warning as Error: `System.Net.Mail.SmtpClient' is obsolete: `SmtpClient and its network of types are poorly designed, we strongly recommend you use https://github.com/jstedfast/MailKit and https://github.com/jstedfast/MimeKit instead'
	Core/Smtp/DefaultSmtpSender.cs(272,16): error CS0618: Warning as Error: `System.Net.Mail.SmtpClient' is obsolete: `SmtpClient and its network of types are poorly designed, we strongly recommend you use https://github.com/jstedfast/MailKit and https://github.com/jstedfast/MimeKit instead'
	Core/Smtp/DefaultSmtpSender.cs(276,25): error CS0618: Warning as Error: `System.Net.Mail.SmtpClient' is obsolete: `SmtpClient and its network of types are poorly designed, we strongly recommend you use https://github.com/jstedfast/MailKit and https://github.com/jstedfast/MimeKit instead'

	 0 Warning(s)
	 6 Error(s)
```

--- a/src/Castle.Core/Core/Smtp/DefaultSmtpSender.cs
+++ b/src/Castle.Core/Core/Smtp/DefaultSmtpSender.cs
@@ -1,4 +1,4 @@
-// Copyright 2004-2009 Castle Project - http://www.castleproject.org/
+// Copyright 2004-2017 Castle Project - http://www.castleproject.org/
 // 
 // Licensed under the Apache License, Version 2.0 (the "License");
 // you may not use this file except in compliance with the License.
@@ -13,6 +13,8 @@
 // limitations under the License.
 
 #if FEATURE_SMTP
+
+#pragma warning disable 618 // Mono marked SmtpClient obsolete
 
 namespace Castle.Core.Smtp
 {
@@ -284,5 +286,7 @@ namespace Castle.Core.Smtp
 		}
 	}
 }
+
+#pragma warning restore 618 // Mono marked SmtpClient obsolete
 
 #endif

